lib/charms/data_platform_libs/v0/data_interfaces.py

```diff
@@ -4316,18 +4388,29 @@ def _on_relation_changed_event(self, event: RelationChangedEvent) -> None:
         if any(newval for newval in diff.added if self.relation_data._is_secret_field(newval)):
             self.relation_data._register_secrets_to_relation(event.relation, diff.added)
 
-        secret_field_user = self.relation_data._generate_secret_field_name(SECRET_GROUPS.USER)
-        if (
-            "username" in diff.added and "password" in diff.added
-        ) or secret_field_user in diff.added:
+        app_databag = get_encoded_dict(event.relation, event.app, "data")
+
+        if app_databag is None:
+            app_databag = {}
```

Comment:
Nit: style preservation with respect to the other RequirerEventHandler classes.

```suggestion
        app_databag = get_encoded_dict(event.relation, event.app, "data")
        if app_databag is None:
            app_databag = {}
```

Reply:
https://github.com/canonical/data-platform-libs/pull/238/commits/5caf34ddd19ec75b197224a302b9832eaf810b7d